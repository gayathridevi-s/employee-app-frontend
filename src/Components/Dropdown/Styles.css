/* label {
    font-size: 16px;
    font-weight: 600;
    color: #b4b7bd;
    position: relative;
    top: -50%;
    left: 15px;
    width: fit-content;
    transition: 0.3s;
    pointer-events: none;
  }
  */
  .input {
    height: 45px;
    width:400px;
    padding-left: 10px;
    border-radius: 5px;
    border: 2px solid #b4b7bd;
    background-color: transparent;
  }
  .dropdownWrap{
    display: flex;
    flex-direction: column;
  }