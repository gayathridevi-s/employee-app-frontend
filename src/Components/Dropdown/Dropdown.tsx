import React, { FC } from 'react';
import './Styles.css';
type DropdownProps = {
  value: string;
  onChange: (value: string) => void;
  label: string;
  placeholder: string;
  options: { value: string; label: string }[]; // Specify the options
};

const Dropdown: FC<DropdownProps> = (props) => {
  return (
    <div className='dropdownWrap'>
      <label>{props.label}</label>
      <select
        className='input'
        value={props.value}
        onChange={(e) => props.onChange(e.target.value)} // Pass the selected value
      >
        <option value='' disabled>
          {props.placeholder}
        </option>
        {props.options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.label}
          </option>
        ))}
      </select>
    </div>
  );
};

export default Dropdown;
