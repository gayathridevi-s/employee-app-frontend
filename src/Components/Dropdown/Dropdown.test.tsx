import React from 'react';
import { render, screen } from '@testing-library/react';
import Dropdown from './Dropdown';

describe('Dropdown Component', () => {
  const mockOptions = [
    { value: 'option1', label: 'Option 1' },
    { value: 'option2', label: 'Option 2' },
    { value: 'option3', label: 'Option 3' }
  ];

  test('renders label and placeholder correctly', () => {
    const props = {
      value: '',
      onChange: jest.fn(),
      label: 'Select an option',
      placeholder: 'Please select',
      options: mockOptions
    };

    render(<Dropdown {...props} />);

    const labelElement = screen.getByText('Select an option');
    const placeholderOption = screen.getByText('Please select');

    expect(labelElement).toBeTruthy();
    expect(placeholderOption).toBeTruthy();
  });

  test('matches snapshot', () => {
    const props = {
      value: '',
      onChange: jest.fn(),
      label: 'Select an option',
      placeholder: 'Please select',
      options: mockOptions
    };

    const { asFragment } = render(<Dropdown {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
