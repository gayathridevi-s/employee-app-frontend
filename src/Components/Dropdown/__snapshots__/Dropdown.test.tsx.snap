// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="dropdownWrap"
  >
    <label>
      Select an option
    </label>
    <select
      class="input"
    >
      <option
        disabled=""
        value=""
      >
        Please select
      </option>
      <option
        value="option1"
      >
        Option 1
      </option>
      <option
        value="option2"
      >
        Option 2
      </option>
      <option
        value="option3"
      >
        Option 3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
